staging/src/k8s.io/apimachinery/pkg/util/validation/validation.go

```diff
@@ -277,6 +277,19 @@ func IsHTTPHeaderName(value string) []string {
 	return nil
 }
 
+const envVarNameFmt = "[-._a-zA-Z][-._a-zA-Z0-9]*"
```

Comment:
> FWIW, ConfigMaps with leading dots are still forbidden.

A config map key may not be `.` or `..` or being with `..`, but leading dots are otherwise allowed